docs/sync.js

```diff
@@ -61,6 +63,44 @@
  * @property {Realm.App.Sync~ClientResetConfiguration|null} [clientReset] - Configuration of Client Reset
  */
 
+/**
+ * Optional object to configure the setup of an initial set of flexible sync
+ * subscriptions to be used when opening the Realm. If this is specified,
+ * {@link Realm.open} will not resolve until this set of subscriptions has been
+ * fully synchronized with the server.
+ *
+ * Example:
+ * ```
+ * const config: Realm.Configuration = {
+ *   sync: {
+ *     user,
+ *     flexible: true,
+ *     initialSubscriptions: {
+ *       update: realm => {
+ *         realm.subscriptions.update(subs => {
+ *           subs.add(realm.objects('Task'));
+ *         })
+ *       }
+ *     }
+ *   },
+ *   // ... rest of config ...
+ * };
+ * const realm = await Realm.open(config);
+ *
+ * // At this point, the Realm will be open with the data for the initial set
+ * // subscriptions fully synchronised.
+ * ```
+ * @typedef {Object} Realm.App.Sync~InitialSubscriptionsConfiguration
+ * @property {callback(realm)} update - callback called with the {@link Realm} instance
+ * to allow you to setup the initial set of subscriptions by calling
+ * `realm.subscriptions.update`. See
+ * {@link Realm.App.Sync.SubscriptionSet#update} for more information.
+ * @property {Boolean} returnOnStartup - optional flag. If `true`, the
```

Comment:
Is this name aligned with the other SDKs? Technically it is re-running when the Realm is opened, right? Not when the app is starting up (although it might if that ends up opening the Realm).

In any case it has a typo.

```suggestion
 * @property {Boolean} rerunOnStartup - optional flag. If `true`, the
```

Reply:
Renamed to `rerunOnOpen`, though will see if other SDKs have a better name